src/modalities/utils/mfu.py

```diff
@@ -0,0 +1,94 @@
+import warnings
+from typing import Optional, Tuple
+
+import torch
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.types import Number
+
+from modalities.util import get_total_number_of_trainable_parameters
+
+# A100: https://developer.nvidia.com/blog/nvidia-ampere-architecture-in-depth/
+# H100: https://developer.nvidia.com/blog/nvidia-hopper-architecture-in-depth/
+#       https://www.nvidia.com/en-us/data-center/h100/
+#
+# NOTE: These values are valid for fp16 and bf16 only
+PEAK_PERFORMANCE = {
+    "A100": 312e12,
+    "H100": 989e12,
+}
+
+
+def _get_theoretical_gpu_peak_performance_single(precision: torch.dtype, gpu_type: str) -> Optional[Number]:
+    """
+    returns theoretical gpu peak performance for #GPU=1 in units FLOPs / s for given gpu type
+    """
+    if precision in [torch.float16, torch.bfloat16] and gpu_type in PEAK_PERFORMANCE.keys():
+        return PEAK_PERFORMANCE[gpu_type]
+    else:
+        return None
+
+
+def get_theoretical_gpu_peak_performance(model: FSDP, world_size: int) -> Optional[Number]:
+    """
+    returns theoretical gpu peak performance for #GPU=world_size in units FLOPs / s for given gpu type
+    """
+    if torch.cuda.is_available() and torch.cuda.device_count() > 0:  # necessary for cpu-only tests
```

Comment:
If torch.cuda.device_count() > 0 is only required for CPU related tests, wouldn't it be prefferable to mock this functions for CPU tests?

Reply:
The function is inherently GPU-related and tested as part of the GPU tests, so I do not see the need to include it in the CPU tests using mocking. The comment "necessary for cpu-only tests" is a bit misleading though: While the `if` statement also makes sure that the CPU tests run through, its main purpose is to avoid the computation and return early & gracefully in the case of no GPUs.

I simply removed the comment in 2708f37c9ca1e0b7d3db64ea52b2370c8d2f7955

Reply:
I see, thanks for the clarification.